enterprise/backend/src/metabase_enterprise/metabot_v3/dummy_tools.clj

```diff
@@ -24,6 +27,73 @@
                {:error "dashboard not found"})
    :context context})
 
+(defn- convert-field-type
+  [db-field]
+  (let [db-field (u/normalize-map db-field)]
+    (cond
+      (lib.types.isa/boolean? db-field)               "boolean"
+      (lib.types.isa/string-or-string-like? db-field) "string"
+      (lib.types.isa/numeric? db-field)               "number"
+      (lib.types.isa/temporal? db-field)              "date")))
+
+(defn- convert-field
+  [db-field]
+  (-> db-field
+      (select-keys [:id :name :description])
+      (update :id #(str "field_" %))
+      (assoc :type (convert-field-type db-field))))
+
+(defn- convert-metric
+  [db-metric]
+  (select-keys db-metric [:id :name :description]))
+
+(declare ^:private table-details)
+
+(defn- foreign-key-tables
+  [fields]
+  (when-let [target-field-ids (->> fields
+                                   (into #{} (keep :fk_target_field_id))
+                                   not-empty)]
+    (->> (t2/select-fn-set :table_id :model/Field :id [:in target-field-ids])
+         (mapv #(table-details % {:include-foreign-key-tables? false})))))
+
+(defn- table-details
+  "Loads the details of a table with ID `id`. If the option `include-foreign-key-tables?` is truthy,
```

Comment:
**question:** This only includes the FK tables that are wired up via semantic type settings, right? So basically the same tables that we could also use for implicit joins in the query-builder.

Reply:
Right.